File: ice0208/hash-map_set/2352.py
class Solution:
    def equalPairs(self, grid: list[list[int]]) -> int:
        s = dict()

        for r in range(len(grid)):
            t = ""
            for c in range(len(grid)):
                t += str(grid[r][c]) + ","
            s[t] = s.get(t, 0) + 1

        answer = 0
        for c in range(len(grid)):
            t = ""
            for r in range(len(grid)):
                t += str(grid[r][c]) + ","
            if t in s:
                answer += s[t]

        return answer


if __name__ == "__main__":
    testcase_list = [
        [[3, 2, 1], [1, 7, 6], [2, 7, 7]],
        [[3, 1, 2, 2], [1, 4, 4, 5], [2, 4, 2, 2], [2, 4, 2, 2]],
    ]

    for testcase in testcase_list:
        print(Solution().equalPairs(grid=testcase))

File: moonpoison/hash_map_set/2352.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int equalPairs(vector<vector<int>>& grid) 
    {
        map<vector<int>, int> m;
        int ans=0;
        for(int i=0;i<grid.size();++i)
           m[grid[i]]++;
        for(int i=0;i<grid.size();++i)
        {
            vector<int> v;
            for(int j=0;j<grid.size();++j)
            {
                v.push_back(grid[j][i]);
            }
            ans+=m[v];
        }
        return ans;
    }
};